### PR TITLE
Establishing information for the vuln db

### PR DESCRIPTION
**Pull request by lirantal**
The following document aims to centralize information for the management and processes involved with maintaining the vulnerability database (for both Core and Ecosystem).

Some points I'd like us to discuss and close down:
1. [x] suggestion to remove the vulnerability database section in the main README (https://github.com/nodejs/security-wg#vulnerability-database) and instead just link for the more verbose part of it in this document
2. [ ] suggestion to remove the markdown at https://github.com/nodejs/security-wg/blob/master/vuln/README.md and instead move that information to this document in the PR
3. [x] Agree on time frame for introducing changes (30 days)
4. [ ] Agree on media for announcement of changes (i.e: Node.js website, or official twitter account)
5. [x] Agree on moving the vuln db to its own repository

### CODE REVIEW

File: processes/vuln_db.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  "cvss_score": 6.5,
+  "coordinating_vendor": null
+}
+```
+
+### Validating reports
+
+The Security WG employs a static code analysis tool that runs in CI for all Pull-Requests to validate the correct structure when new vulnerabilities are introduced.
+
+You may make use of the same tool which is available [here](https://github.com/nodejs/security-wg/blob/master/tools/vuln_valid/index.js)
```
Comment (vdeturckheim) [inline processes/vuln_db.md]: Can we mention that it is also possible to run this tool using `npm test` ?

File: processes/vuln_db.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+We will announce an upcoming change in the following ways:
+- A new issue in the repository (see example reference: https://github.com/nodejs/security-wg/issues/345)
+- We will contact the list of known database users (see section below) and alert them of the change, inviting them to participate in the discussion before changes are merged.
+- By social media, using twitter, to further broadcast the change through some of the Security WG's twitter accounts.
+
+Time window for introducing changes:
+- Changes that affect existing database fields will provide a 30 days time-frame through announcement for any party to chime in and weigh on the changes proposed
+- Changes that extend the database fields with newly proposed fields will be added as the proposed Pull-Request is merged
+
+Note: we recommended to subscribe to the repository's issue queue in order to receive updates in a timely manner for both new issues that may influence database changes, as well as new vulnerabilities pushed to the database or changes made to existing.
```
Comment (vdeturckheim) [inline processes/vuln_db.md]: Also, we have a flag for everything relative to vulnerabilities

**Comment by vdeturckheim** [comment]
> suggestion to remove the vulnerability database section in the main README (https://github.com/nodejs/security-wg#vulnerability-database) and instead just link for the more verbose part of it in this document

LGTM!

> suggestion to remove the markdown at https://github.com/nodejs/security-wg/blob/master/vuln/README.md and instead move that information to this document in the PR

I like having the README.md as this is the file Github displays when people browse the repo. This is where I would be expecting to see information regarding the DB.

> Agree on time frame for introducing changes

30 days lgtm

> Agree on media for announcement of changes (i.e: Node.js website, or official twitter account)

LGTM

Side note, I'm thinking about updating our CI job to upload the DB to Algolia and provide a free to use way to search through the dataset. Would it make sense?

File: processes/vuln_db.md [hunk 3]
```diff
@@ -72,8 +72,7 @@ Time window for introducing changes:
 - Changes that affect existing database fields will provide a 30 days time-frame through announcement for any party to chime in and weigh on the changes proposed
 - Changes that extend the database fields with newly proposed fields will be added as the proposed Pull-Request is merged
 
-Note: we recommended to subscribe to the repository's issue queue in order to receive updates in a timely manner for both new issues that may influence database changes, as well as new vulnerabilities pushed to the database or changes made to existing.
-
+Note: we recommended to subscribe to the repository's issue queue in order to receive updates in a timely manner for both new issues that may influence database changes, as well as new vulnerabilities pushed to the database or changes made to existing. Also, all Pull-Request that add a new vulnerability are labled with `vulnerabilities` which makes them easily filtered and searchable.
```
Comment (cjihrig) [inline processes/vuln_db.md]: issue queue -> issue tracker (I know that's not really part of this change)
Pull-Request -> pull request

**Comment by cjihrig** [comment]
I don't have a strong opinion.

**Comment by reedloden** [comment]
Might I recommend that the vuln db be moved to a separate repo than this one? Basically, would be nice to have something similar to https://github.com/rubysec/ruby-advisory-db/ that is just the vulnerability data only.

**Comment by pxlpnk** [comment]
I like what @reedloden suggested. This would reduce the noise on this repo and allow consumers of the data focus on the essential notifications.

As for the READMEs, I think what @vdeturckheim said makes sense. Especially if we decide now or later to move the data into its own repository.

**Comment by drifkin** [comment]
I also think moving the advisories to a separate repo would be great, right now it is a little difficult to follow this repo.

File: processes/vuln_db.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+We will announce an upcoming change in the following ways:
+- A new issue in the repository (see example reference: https://github.com/nodejs/security-wg/issues/345)
+- We will contact the list of known database users (see section below) and alert them of the change, inviting them to participate in the discussion before changes are merged.
+- By social media, using twitter, to further broadcast the change through some of the Security WG's twitter accounts.
+
+Time window for introducing changes:
+- Changes that affect existing database fields will provide a 30 days time-frame through announcement for any party to chime in and weigh on the changes proposed
+- Changes that extend the database fields with newly proposed fields will be added as the proposed pull request is merged
+
+Note: we recommended to subscribe to the repository's issue tracker in order to receive updates in a timely manner for both new issues that may influence database changes, as well as new vulnerabilities pushed to the database or changes made to existing. Also, all pull request that add a new vulnerability are labled with `vulnerabilities` which makes them easily filtered and searchable.
```
Comment (mhdawson) [inline processes/vuln_db.md]: we recommend subscribing to the ...

Comment (mhdawson) [reply processes/vuln_db.md]: or changes made to existing **entries.**

Comment (mhdawson) [reply processes/vuln_db.md]: Also, all pull request _> Also, all pull request**s**

File: processes/vuln_db.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+
+## Database Users
+
+The following is a list of users or organizations which opt-in to be listed as users of the vulnerability database:
+
+- [ ] Customer 1 (contact: customer@example.com)
```
Comment (mhdawson) [inline processes/vuln_db.md]: I'll be interested to see if people are willing to list emails in a public doc.  We may need these to be github ids, but lest see what people think.

**Comment by mhdawson** [comment]
In respect to:

suggestion to remove the vulnerability database section in the main README  (https://github.com/nodejs/security-wg#vulnerability-database) and instead just link for the more verbose part of it in this document
+1 from me

  suggestion to remove the markdown at https://github.com/nodejs/security-wg/blob/master/vuln/README.md and instead move that information to this document in the PR
+1

  Agree on time frame for introducing changes
I think the 30 days for comment and then 30 days notice before the change actually occurs seems like it would be reasonable.

  Agree on media for announcement of changes (i.e: Node.js website, or official twitter account)
The more the better :)

  Agree on moving the vuln db to its own repository
+    1 from me

File: processes/vuln_db.md (hunk 4 above)
Comment (lirantal) [reply processes/vuln_db.md]: thanks, fixed!

File: processes/vuln_db.md (hunk 5 above)
Comment (lirantal) [reply processes/vuln_db.md]: right. I was thinking some of these will actually be organizations like snyk, etc so you'd probably have an official e-email there.

**Comment by lirantal** [comment]
@mhdawson I updated the time window to be 60 days - 30 days to request for comments, and then another 30 days of announcements before the change is commited.

**Comment by lirantal** [comment]
@vdeturckheim @mhdawson do you want to review for any other changes or should we go ahead and land this?

**Pull request merged**